beacon-chain/cache/depositcache/deposits_cache.go

```diff
@@ -248,6 +248,10 @@ func (dc *DepositCache) FinalizedDeposits(ctx context.Context) *FinalizedDeposit
 	dc.depositsLock.RLock()
 	defer dc.depositsLock.RUnlock()
 
+	if dc.finalizedDeposits == nil {
```

Comment:
I dont think this is the right solution, this just puts a band-aid over the original error. Why is there a nil finalized deposits cache in the first place ?

Reply:
Thank you for pointing this out. I did not notice the `NewDepositCache()` constructor function. Taking advantage of the constructor allowed me to simplify the code.

I am a bit unsure about the `finalizedDepositsTrie, MerkleTrieIndex: -1` initialization. Is this acceptable?

Reply:
That is the correct behaviour, the first deposit will lead to an index of `0` which is correct.